src/aggregators/ccnrm.jl

```diff
@@ -0,0 +1,158 @@
+    # execute jump
+    u = update_state!(p, integrator, u, affects!)
+
+    # update current jump rates and times
+    update_dependent_rates!(p, u, params, t)
+    nothing
+end
+
+# calculate the next jump / jump time
+# just the first reaction in the first non-empty bin in the priority table
+function generate_jumps!(p::CCNRMJumpAggregation, integrator, u, params, t)
+    next_jump, next_time = getfirst(p.ptt)
+
+    # Rebuild the table if no next jump is found. 
+    if next_jump === nothing
+        binwidth = 16 / sum(p.cur_rates)
+        min_time = minimum(p.ptt.times)
+        rebuild!(p.ptt, min_time, binwidth)
+        next_jump, next_time = getfirst(p.ptt)
+    end
+
+    p.next_jump, p.next_jump_time = next_jump, next_time
+    nothing
+end
+
+######################## SSA specific helper routines ########################
+
+# Recalculate jump rates for jumps that depend on the just executed jump (p.next_jump)
+function update_dependent_rates!(p::CCNRMJumpAggregation, u, params, t)
+    @inbounds dep_rxs = p.dep_gr[p.next_jump]
+    @unpack ptt, cur_rates, rates, ma_jumps, end_time = p
+    num_majumps = get_num_majumps(ma_jumps)
+
+    @inbounds for rx in dep_rxs
+        oldrate = cur_rates[rx]
+        times = ptt.times
+        oldtime = times[rx]
+
+        # update the jump rate
+        @inbounds cur_rates[rx] = calculate_jump_rate(ma_jumps, num_majumps, rates, u,
+            params, t, rx)
+
+        # Calculate new jump times for dependent jumps
+        if rx != p.next_jump && oldrate > zero(oldrate)
+            if cur_rates[rx] > zero(eltype(cur_rates))
+                update!(ptt, rx, oldtime, t + oldrate / cur_rates[rx] * (times[rx] - t))
+            else
+                update!(ptt, rx, oldtime, 2 * end_time)
+            end
+        else
+            if cur_rates[rx] > zero(eltype(cur_rates))
+                update!(ptt, rx, oldtime, t + randexp(p.rng) / cur_rates[rx])
+            else
+                update!(ptt, rx, oldtime, 2 * end_time)
+            end
+        end
+    end
+    nothing
+end
+
+# Evaluate all the rates and initialize the times in the priority table. 
+function initialize_rates_and_times!(p::CCNRMJumpAggregation, u, params, t)
+    # Initialize next-reaction times for the mass action jumps
+    majumps = p.ma_jumps
+    cur_rates = p.cur_rates
+    pttdata = Vector{typeof(t)}(undef, length(cur_rates))
+    @inbounds for i in 1:get_num_majumps(majumps)
+        cur_rates[i] = evalrxrate(u, i, majumps)
+        pttdata[i] = t + randexp(p.rng) / cur_rates[i]
+    end
+
+    # Initialize next-reaction times for the constant rates
+    rates = p.rates
+    idx = get_num_majumps(majumps) + 1
+    @inbounds for rate in rates
+        cur_rates[idx] = rate(u, params, t)
+        pttdata[idx] = t + randexp(p.rng) / cur_rates[idx]
+        idx += 1
+    end
+
+    # Build the priority time table with the times and bin width. 
+    binwidth = 16 / sum(cur_rates)
```

Comment:
Do not hardcode `16` everywhere. Make it a global constant or a field in the aggregator